### PR TITLE
docs: Add parsing JSON mixed-up with plain text log message

### PR DESCRIPTION
**Pull request by danybmx**
* What problems does this PR solve?
- We're adding a new feature to logs parsing that allow to parse grok matches to JSON, here we try to explain how to do it and the configurations available.

* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.
Added section is in a Collapser because it's related to the other ones from the collapser. The `json` type is added as well to the grok-types table because it's not a type as that, but the way to use those "actions" is the same as for types in Grok, and also, it could be understandable as a "type casting" as the others, you're casting your "text" into a "structured json" type.

* If your issue relates to an existing GitHub issue, please link to it.
No

### CODE REVIEW

File: src/content/docs/logs/ui-data/parsing.mdx [hunk 1]
```diff
@@ -289,9 +289,52 @@ Watch our YouTube video (approx. 6 minutes) to learn how to use a Grok debugger
...
         </tbody>
       </table>
   </Collapser>
+  <Collapser
+    id="parsing-json"
+    title="Parsing JSON mixed-up with plain text"
+  >
+    New Relic logs pipeline parses your log JSON messages by default but sometimes you've JSON log messages that are mixed up with plain text and in that case you also want to be able to parse them and then be able to filter using the JSON attributes.
+
+    If thats the case you can use the `json` [grok type](#grok-syntax) that will do exactly that, parse the JSON extracted by the grok pattern.
```
Comment (jdearana-nr) [inline src/content/docs/logs/ui-data/parsing.mdx:310]:
```suggestion
    If that is the case, you can use the `json` [grok type](#grok-syntax), which will parse the JSON captured by the grok pattern.
```

File: src/content/docs/logs/ui-data/parsing.mdx [hunk 2]
```diff
@@ -289,9 +289,52 @@ Watch our YouTube video (approx. 6 minutes) to learn how to use a Grok debugger
...
+
+    As an example, imagine your logs are JSON logs but prefixed with a date time string:
```
Comment (jdearana-nr) [inline src/content/docs/logs/ui-data/parsing.mdx:312]:
```suggestion
    Using the `json` [grok type](#grok-syntax), you can extract and parse JSON from logs that are not properly formatted, e.g, your logs are prefixed with a date time string:
```

File: src/content/docs/logs/ui-data/parsing.mdx [hunk 3]
```diff
@@ -289,9 +289,52 @@ Watch our YouTube video (approx. 6 minutes) to learn how to use a Grok debugger
...
+
+    ```
+    2015-05-13T23:39:43.945958Z {"event": "TestRequest", "status": 200, "response": {"headers": {"X-Custom": "foo"}, "request": {"headers": {"X-Custom": "bar"}}}}
+    ```
+
+    For parsing that we can create a Parsing Rule with the following Grok:
```
Comment (jdearana-nr) [inline src/content/docs/logs/ui-data/parsing.mdx:318]:
```suggestion
    In order to extract and parse the JSON data from this log format, you would create the following Grok expression:
```

File: src/content/docs/logs/ui-data/parsing.mdx [hunk 4]
```diff
@@ -289,9 +289,52 @@ Watch our YouTube video (approx. 6 minutes) to learn how to use a Grok debugger
...
+
+    ```
+    %{TIMESTAMP_ISO8601:containerTimestamp} %{GREEDYDATA:log:json}
+    ```
+
+    And the log will result as:
```
Comment (jdearana-nr) [inline src/content/docs/logs/ui-data/parsing.mdx:324]:
```suggestion
    The resulting log would be:
```

File: src/content/docs/logs/ui-data/parsing.mdx [hunk 5]
```diff
@@ -289,9 +289,52 @@ Watch our YouTube video (approx. 6 minutes) to learn how to use a Grok debugger
...
+    ```
+    containerTimestamp: "2015-05-13T23:39:43.945958Z"
+    log.event: "TestRequest"
+    log.status: 200
+    log.response.headers.X-Custom: "foo"
+    log.request.headers.X-Custom: "bar"
+    ```
+
+    It is possible to configure some JSON parser behaviors using `:json(_CONFIG_)`:
```
Comment (jdearana-nr) [inline src/content/docs/logs/ui-data/parsing.mdx:333]:
```suggestion
    Optionally, you can configure the `json` [grok type](#grok-syntax) using `:json(_CONFIG_)`:
```

File: src/content/docs/logs/ui-data/parsing.mdx [hunk 6]
```diff
@@ -289,9 +289,52 @@ Watch our YouTube video (approx. 6 minutes) to learn how to use a Grok debugger
...
+
+    - `json({"dropOriginal": true})`: Drop original field, this defaults to true and means that will drop the JSON string attribute after parsing it.
```
Comment (jdearana-nr) [inline src/content/docs/logs/ui-data/parsing.mdx:335]:
```suggestion
    - `json({"dropOriginal": true})`: Drop the original attribute. When set to `true` (default value), the Parsing rule will drop the original JSON attribute.
```

File: src/content/docs/logs/ui-data/parsing.mdx [hunk 7]
```diff
@@ -289,9 +289,52 @@ Watch our YouTube video (approx. 6 minutes) to learn how to use a Grok debugger
...
+    - `json({"depth": 62})`: Allows to control how much levels you want to parse from your JSON (It defaults to its max that is 62).
```
Comment (jdearana-nr) [inline src/content/docs/logs/ui-data/parsing.mdx:336]:
```suggestion
    - `json({"depth": 62})`: Levels of depth you want to parse the JSON value (defaulted to 62).
```

**Review by jdearana-nr — COMMENTED** [review]
Some suggestions. TY!

**Review by barbnewrelic — APPROVED** [review]
Thanks for adding this info about JSON + plain text!

**Pull request merged**